pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -141,9 +141,10 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 				}
 			}
 			selectedNodes := util.SelectBestNode(nodeScores)
+			req := api.GetPodResourceRequest(task.Pod)
```

Comment:
`task.Resreq` already includes `initContainer`, why do we need to re-calculate it again?

Reply:
`task.Resreq` is calculated by `GetPodResourceWithoutInitContainers`, it does not contain `initContainer`. We only need consider `initContainer` when `Allocate`, do I understand it correctly?

Reply:
We also need Resreq in reclaim, preemption and backfill. And predicates&priorities will use Pods instead of task info :)

Reply:
Addressed. Thanks!

Reply:
hm.... can we update `task.Resreq` directly, so we do not need to update different actions :)

Reply:
I updated it at first time. However a lot of functions(e.g. `jobinfo.AddTaskInfo`, `NodeInfo.AddTaskInfo`) will use `task.Resq` that does not contain `init container`. Maybe we could add another variable `ResqWithInitContainers` in `TaskInfo`?

Reply:
If so, prefer to update related code to consider `initContainer` :)